resolver/resolver.go

```diff
@@ -330,3 +331,19 @@ type AuthorityOverrider interface {
 	// typically in line, and must keep it unchanged.
 	OverrideAuthority(Target) string
 }
+
+// ValidateEndpoints validates endpoints are valid from a petiole policies
```

Comment:
"Valid" now appears 3 times.  And spelling mistake:
`ValidateEndpoints validates endpoints from a petiole _policy's_ perspective.`

But now you should also explain what that actually means?

How about:

> ValidateEndpoints validates that the endpoints are valid to use in a petiole policy - i.e. they must be non-empty and each must contain at least one address.

And what's a petiole policy? Is this defined somewhere?  Maybe add: "See [A61] for details"?

Reply:
Hmmm ok. At first I thought I needed to add Outlier Detection to this, but that isn't a petiole policy (can't use it directly on top of pick first, that is invalid), and this is meant to trigger reresolution and reject the update. This happens from the petiole policy layer so the only thing important in OD is to ignore endpoints with no addresses and to have the undefined behavior in shared addresses, and leave petiole policies to trigger this error.

Reply:
Switched to

```
// ValidateEndpoints validates endpoints from a petiole policy's perspective.
// Petiole policies should call this before calling into their children. See
//[gRPC A61](https://github.com/grpc/proposal/blob/master/A61-IPv4-IPv6-dualstack-backends.md)
// for details.
```